robot_modeling/DQ_SerialManipulatorDH.m

```diff
@@ -19,14 +19,16 @@
 % second row of A will have the joints offsets.
 %
 % DQ_SerialManipulatorDH Methods (Concrete):
-%       get_dim_configuration_space - Return the dimension of the configuration space.
-%       fkm - Compute the forward kinematics while taking into account base and end-effector's rigid transformations.
-%       plot - Plots the serial manipulator.
-%       pose_jacobian - Compute the pose Jacobian while taking into account base's and end-effector's rigid transformations.
+%       get_thetas - Returns the first row of the DH Matrix ('theta' parameter). 
```

Comment:
Hello @juanjqo,

I agree because we need to maintain consistency. In any case, because a robot modeled using DH parameters can have either revolute or prismatic joints, $\theta_i$ and $d_i$ can be either a variable or a parameter. Do we have a function that gives us the configuration vector?

Reply:
Hi @bvadorno, as far as I know, we do not have such a function. I know that in the superclass DQ_Kinematics we have the protected property _q_. But we are not using that property (at least in the classes related to serial manipulators). The configuration vector is used as an argument in some methods.

Reply:
In that case, proceed with my previous suggestion. If we didn't need an explicit get method to retrieve the robot configuration until now, implementing it is not urgent.

Reply:
@juanjqo @bvadorno 

For either revolute or prismatic joints, the $\theta_i$ or $d_i$ are the offsets of the respective DH parameter. The effective displacement will be $\theta_i+q_i$ or $d_i + q_i$ depending on the joint type.
Until now, the configuration vector has never been stored in the kinematics object. It has always been an argument for the method in question.

@juanjqo changed the internal implementation in MATLAB to look like CPP (I guess to make the adjustments easier?), so some changes got a bit difficult for us to track.

@bvadorno Up until the last release version, `a`, `alpha` etc were class properties without access control
https://github.com/dqrobotics/matlab/blob/f11f0de06784d04042b8713010373d08fb86ccf8/robot_modeling/DQ_SerialManipulator.m#L55
so the user could freely get and set the values of the DH parameters, such as `a` with `robot.a(2)`.

That was (probably?) unintended, so we don't necessarily need to give support to that anymore. This is also where the `get_as()` came from. `get` because we introduced getters and plural because it does not return a single number.

If 
`double DQ_SerialManipulatorDH::get_a(int)`
`VectorXd DQ_SerialManipulatorDH::get_as()` 
are not informative enough, we can do something like:
`double DQ_SerialManipulatorDH::get_dh_parameter_a(int) const`
`VectorXd DQ_SerialManipulatorDH::get_dh_parameter_as() const`

The `_vector()` suffix might also be understood as an implicit commitment to (re)naming methods that return vectors in with the same suffix. In other libraries and interfaces, this is usually dealt with by using plural, for instance `singularValues` in Eigen3
https://eigen.tuxfamily.org/dox/classEigen_1_1JacobiSVD.html
Or `get_joint_positions()` in `DQ_VrepInterface` etc.

 Anyways IMO it is important to be clear and not use plural in a method signature if it is not returning a vector.
 `VectorXd get_parameter_a()` which returns all `a`s seems confusing to me.

Reply:
@bvadorno @mmmarinho I added two new methods trying to follow the @mmmarinho's suggestion. To do so, I added a new property:
```Matlab
    properties (Constant)
        parameter = struct('theta', 1, 'd',2, 'a', 3, 'alpha', 4, 'JointType', 5);
    end
```
The methods look like this:
```Matlab
        function ret = get_parameters(obj, parameter)
            %GET_PARAMETERS(parameter) Returns the vector containing the
            % parameter specified. The parameter can be theta, a, d, alpha, 
            % and JointType
            ret = obj.dh_matrix_(parameter,:);
        end

        function ret = get_parameter(obj, parameter, ith)
            %GET_PARAMETERS(parameter, ith) Returns the scalar containing the
            % parameter specified of the ith joint. 
            % The parameter can be theta, a, d, alpha, and JointType
            ret = obj.dh_matrix_(parameter,ith);
        end
```
Usage:
```Matlab
robot = DQ_SerialManipulatorDH(ax18_DH_matrix); % Defines robot model using dual quaternions

robot.get_parameters(robot.parameter.a)
robot.get_parameter(robot.parameter.a, 3)

robot.get_parameters(robot.parameter.alpha)
robot.get_parameter(robot.parameter.alpha, 2)
```
What do you think about this?

Reply:
@mmmarinho @bvadorno Using the same strategy to implement the get_parameters method, I added a new struct property called JointType. We have now the following constant properties
```Matlab
    properties (Constant)
        % Joints that can be actuated
        % Revolute joint
        JOINT_ROTATIONAL = 1; % Deprecated
        % Prismatic joint
        JOINT_PRISMATIC = 2;   % Deprecated
   
        Parameter = struct('Theta', 1, 'D',2, 'A', 3, 'Alpha', 4, 'JointType', 5);    % New
        JointType = struct('REVOLUTE',1, 'PRISMATIC',2);                              % New
    end
```
Usage:

```Matlab
ax18_DH_theta   = [0       0      -pi/2   -pi/2      -pi/2];  % theta
            ax18_DH_d       = [0.167   0       0     0.1225          0];  % d
            ax18_DH_a       = [0      0.159    0.02225    0      0.170];  % a
            ax18_DH_alpha   = [-pi/2   0      -pi/2   -pi/2          0];  % alpha
            ax18_DH_type    = [repmat(DQ_SerialManipulatorDH.JointType.REVOLUTE,1,5)];
            
            ax18_DH_matrix = [  ax18_DH_theta;
                                               ax18_DH_d;
                                               ax18_DH_a;
                                               ax18_DH_alpha;
                                               ax18_DH_type;
               ]; 
    
robot = DQ_SerialManipulatorDH(ax18_DH_matrix);
```